### PR TITLE
Change maintainer label to opencontainers in Dockerfile

### PR DESCRIPTION
**Pull request by mauri870**
## Proposed commit message

The old way of specifying maintainers in a Dockerfile was to use the MAINTAINER instruction. This is now deprecated and the maintainer label was used instead.

Nowadays the recommended way to set a maintainer is through the `org.opencontainers.image.authors` label.

See https://docs.docker.com/reference/build-checks/maintainer-deprecated/.

### CODE REVIEW

**Review by swiatekm — COMMENTED** [review]
Shouldn't this merit a changelog entry? It is a user-facing change, if a small one.

**Comment by mauri870** [comment]
> Shouldn't this merit a changelog entry? It is a user-facing change, if a small one.

I had considered this when making the same change in Beats (https://github.com/elastic/beats/pull/40736). While I agree that it is a user-facing change, I highly doubt any tool would rely on a maintainer label, especially since it's not standardized anywhere and the value is a free form string. That said, I could be wrong, so I'll wait for someone with more experience to weigh in.

**Comment by swiatekm** [comment]
> I had considered this when making the same change in Beats (https://github.com/elastic/beats/pull/40736). While I agree that it is a user-facing change, I highly doubt any tool would rely on a maintainer label, especially since it's not standardized anywhere and the value is a free form string. That said, I could be wrong, so I'll wait for someone with more experience to weigh in.

Are there any downsides to including a changelog entry here?

**Review by michel-laterman — APPROVED** [review]

**Comment by mauri870** [comment]
> Are there any downsides to including a changelog entry here?

I guess there is not, I have added a changelog entry.

**Review by swiatekm — APPROVED** [review]

**Comment by mauri870** [comment]
The build failure in k8s tests seems to be unrelated, I'm merging the changes.

**Comment by pierrehilbert** [comment]
The failures seem to be related to the 9.0 version bump preparation.

**Comment by mauri870** [comment]
> The failures seem to be related to the 9.0 version bump preparation.

Thanks Pierre, should we wait for the tests to be passing on main or can you force merge this change?

**Comment by pierrehilbert** [comment]
I don't think we have a huge pressure about merging this PR so probably best to wait and avoid abusing the force merge option. WDYT?

**Comment by pierrehilbert** [comment]
@mauri870 shouldn't we backport to 8.x branch?

**Comment by mauri870** [comment]
> @mauri870 shouldn't we backport to 8.x branch?

Yes, sorry. I went through my PRs adding the backport label, seems like I missed this one.

**Pull request merged**
